staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -1086,6 +1086,24 @@ func TestPriorityAndFairnessWithPanicRecoveryAndTimeoutFilter(t *testing.T) {
 	})
 }
 
+func TestDisabledApfWatermarkMainteneace(t *testing.T) {
+	server := newApfServerWithSingleRequest(t, decisionQueuingExecute, false)
+	defer server.Close()
+
+	// when we disable watermarks that mean that we do not write to waitingMark variable
```

Comment:
I don't think that setting waitingMark to nil here is testing what it says it does. It looks like waitingMark is only used when you invoke newApfServerWithSingleRequest with "true" as the last argument instead of "false" (if it is set to true, then you get  panic if you set waitingMark to nil).

Reply:
fixed in 8f83617cbe4ec32dedc6c185ce011814cb8ea0b5